src/plugins/vis_default_editor/public/components/agg_param_props.ts

```diff
@@ -46,3 +51,9 @@ export interface AggParamEditorProps<T, P = AggParam> extends AggParamCommonProp
   setValidity(isValid: boolean): void;
   setTouched(): void;
 }
+
+export interface OptionedParamEditorProps<T = OptionedValueProp> {
+  aggParam: {
+    options: T[];
+  };
+}
```

Comment:
Moved this out of `data` and into the editor, as it felt UI specific and said `Editor`